test/e2e/tests/multichain/connection-page.spec.js

```diff
@@ -0,0 +1,141 @@
+        await driver.clickElement('button[aria-label="Back"]');
+        await driver.clickElement('button[aria-label="Back"]');
+        // validate dapp is not connected
+        await driver.clickElement(
+          '[data-testid ="account-options-menu-button"]',
+        );
+        await driver.clickElement({ text: 'All Permissions', tag: 'div' });
+        const noAccountConnected = await driver.isElementPresent({
+          text: 'Nothing to see here',
+          tag: 'p',
+        });
+        assert.ok(
+          noAccountConnected,
+          'Account disconected from connections page',
+        );
+      },
+    );
+  });
+
+  it('should connect more accounts when already connected to a dapp', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder().build(),
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+        if (!process.env.MULTICHAIN) {
+          return;
+        }
+        await waitForAccountRendered(driver);
+        await connectToDapp(driver);
+
+        // close test dapp window to avoid future confusion
+        const windowHandles = await driver.getAllWindowHandles();
+        await driver.closeWindowHandle(windowHandles[1]);
+        // disconnect dapp in fullscreen view
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        // Add new account with custom label
+        await driver.clickElement('[data-testid="account-menu-icon"]');
+        await driver.clickElement(
+          '[data-testid="multichain-account-menu-popover-action-button"]',
+        );
+        await driver.clickElement(
+          '[data-testid="multichain-account-menu-popover-add-account"]',
+        );
+        await driver.fill('[placeholder="Account 2"]', anotherAccountLabel);
+        await driver.clickElement({ text: 'Create', tag: 'button' });
+        await waitForAccountRendered(driver);
+        await driver.clickElement(
+          '[data-testid ="account-options-menu-button"]',
+        );
+        await driver.clickElement({ text: 'All Permissions', tag: 'div' });
+        await driver.clickElement({ text: 'Got it', tag: 'button' });
+        await driver.clickElement({
+          text: '127.0.0.1:8080',
+          tag: 'p',
+        });
+
+        await driver.clickElement({
+          text: 'Connect more accounts',
+          tag: 'button',
+        });
+        await driver.clickElement({
+          text: '2nd custom name',
+          tag: 'button',
+        });
+        await driver.clickElement(
+          '[data-testid ="connect-more-accounts-button"]',
+        );
+        const newAccountConnected = await driver.isElementPresent({
+          text: '2nd custom name',
+          tag: 'button',
+        });
+
+        assert.ok(newAccountConnected, 'Connected More Account Successfully');
```

Comment:
Likewise, can we switch back to the test dapp and ensure that ETH_ACCOUNTS lists two accounts?

I also think that creating 3 accounts and checking that only 2 are connected would help.

Reply:
Done! I have used the account addresses to verify which accounts are connected in Dapp here https://github.com/MetaMask/metamask-extension/pull/24161/commits/91a93c0d5bd64cb8ff17b06f640cc76f44ae7903